packages/dds/tree/src/feature-libraries/schemaSummarizer.ts

```diff
@@ -170,6 +170,25 @@ export class SchemaEditor<TRepository extends StoredSchemaRepository>
 		return false;
 	}
 
+	/**
+	 * @returns true iff this is a schema op and was submitted.
+	 *
+	 * TODO: Shared tree needs a pattern for handling non-changeset operations.
+	 * See TODO on `SharedTree.processCore`.
+	 */
+	public tryResubmitOp(content: JsonCompatibleReadOnly): boolean {
+		const op: JsonCompatibleReadOnly = content;
+		if (isJsonObject(op) && op.type === "SchemaOp") {
+			const schemaOp: SchemaOp = {
+				type: op.type,
+				data: op.data as string,
+			};
+			this.submit(schemaOp);
```

Comment:
This is a little odd. Why copy the op here?
Also it would be better to have a type check that will actually error if its violated rather than using "as".

If you are just going to "as" cast anyway, might as well do:
```suggestion
			this.submit(schemaOp as SchemaOp);
```

Its just as safe as what you have, but simpler.

I don't know if it builds, ( I guess not, index signatures since don't narrow well) but if this builds it would be better:

```suggestion
			assert(typeof op.data === 'string', "expected string data for resubmitted schema op");
			this.submit(schemaOp);
```

If you can't make something like that work,

```suggestion
			assert(typeof op.data === 'string', "expected string data for resubmitted schema op");
			const schemaOp: SchemaOp = {
				type: op.type,
				data: op.data,
			};
			this.submit(schemaOp);
```

would be preferred (when code has no performance concerns, use a check that errors at runtime if violated, especially when this could be caused by misuse of public API surfaces or invalid persisted data instead of just internal bugs).

Also "as" casts are pretty unsafe and fragile. Refactoring's a more likely to compile when they shouldn't if there are "as" casts than with asserts.

Reply:
updated typecast with third suggestion